mlflow/tracking/client.py

```diff
@@ -972,11 +992,12 @@ def log_text(self, run_id, text, artifact_file) -> None:
                 f.write(text)
 
     @experimental
-    def log_dict(self, run_id, dictionary, artifact_file) -> None:
+    def log_dict(self, run_id: str, dictionary: Any, artifact_file: str) -> None:
         """
-        Log a dictionary as an artifact. The serialization format (JSON or YAML) is automatically
-        inferred from the extension of `artifact_file`. If the file extension doesn't exist or
-        match any of [".json", ".yml", ".yaml"], JSON format is used.
+        Log a JSON/YAML-serializable object (e.g. `dict`) as an artifact. The serialization
+        format (JSON or YAML) is automatically inferred from the extension of `artifact_file`.
+        If the file extension doesn't exist or match any of [".json", ".yml", ".yaml"],
+        JSON format is used.
```

Comment:
Fixed the docstring  because `log_dict` works for any JSON/YAML-serializable object (@WeichenXu123's suggestion)